### ISSUE TITLE
Design a simple online update system for the runtime itself

### ISSUE THREAD

**Issue by mikehearn** [mikehearn/graviton-browser#4]
This is _not_ update for apps, but rather for the runtime itself. It should be able to:

1. Download the latest version of itself, with resumption of interrupted downloads
2. Verify a signature over the contents, preferably multiple signatures for multi-sig reproducible updates
3. Unpack alongside the currently running version with a new version number, so it'll be used next time the app starts.
4. A few days successful update, delete older versions of the app to free up disk space.
5. The server should be able to start indicating that the old version should be used again, for quick rollback of botched updates. In that case a "pin file" should be created that tells the bootstrap tool to not use the latest version, but rather, an older one.

**Comment by mikehearn** [comment]
Implemented for Mac. Waiting on further testing, some small improvements and Windows support.

**Comment by mikehearn** [comment]
Works on macOS, however, old versions are not currently deleted and there's no pinning mechanism.

Will consider this done and open a new ticket for Windows testing and bugfixing.
